pyop2/codegen/builder.py

```diff
@@ -442,6 +453,91 @@ def emit_unpack_instruction(self, *, loop_indices=None):
         yield call
 
 
+class MixedMatPack(Pack):
+
+    insertion_names = {False: "MatSetValuesBlockedLocal",
+                       True: "MatSetValuesLocal"}
+    """Function call name for inserting into the PETSc Mat. The keys
+       are whether or not maps are "unrolled" (addressing dofs) or
+       blocked (addressing nodes)."""
+
+    def __init__(self, packs, access, dtype, block_shape, interior_horizontal=False):
+        self.access = access
+        assert len(block_shape) == 2
```

Comment:
So we cannot assemble mixed vectors? Do we have something like `MixedVecPack`s? Or is the shape here something like `block_shape =(n,1) `? Just asking because in Slate it would be `block_shape = (n,)`, I think.

Reply:
There's a `MixedDatPack` a bit further up.